lib/streamlit/web/server/routes.py

```diff
@@ -213,7 +213,12 @@ async def get(self) -> None:
             # disallows writing lists directly into responses due to potential XSS
             # vulnerabilities.
             # See https://www.tornadoweb.org/en/stable/web.html#tornado.web.RequestHandler.write
-            self.write({"allowedOrigins": ALLOWED_MESSAGE_ORIGINS})
+            self.write(
+                {
+                    "allowedOrigins": ALLOWED_MESSAGE_ORIGINS,
+                    "useExternalAuthToken": False,
```

Comment:
Is this `useExternalAuthToken` would be converted to a config option or something similar?
I am wondering how cloud now could set "useExternalAuthToken" to True?

Reply:
It'll be set to true in Cloud/SiS/other deployment environments by either intercepting requests to the `/st-allowed-message-origins` endpoint or because the OS Tornado server isn't used at all in the environment, so we likely don't ever have to worry about providing a way to set this to true within the OS lib itself.